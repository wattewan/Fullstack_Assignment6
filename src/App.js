import React from "react";
import './App.css'
import { Route, Switch } from "react-router-dom";
import Home from './components/Home';
import About from './components/About';
import Error from './components/Error';
import Navbar from './components/Navbar';
import ThemeSwitcher from "./components/ThemeSwitcher";
import API from './components/Api';





function App() {
  return (
    <div className="todoapp stack-large">
      <ThemeSwitcher />
      <Navbar />
      
      <Switch>
        <Route path="/" component={Home} exact /> |
        <Route path="/about" component={About} />
        <Route path="/api" component={API} />
        <Route component={Error} />
      </Switch>
    </div>
  )
}

// class AppTheme extends React.Component {
//   render() {
//     return (
//       <ThemeContext.Provider value='dark'>
//         <Toolbar />
//       </ThemeContext.Provider>
//     )
//   }
// }




export default App;