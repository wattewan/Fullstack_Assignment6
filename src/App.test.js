import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Form from './components/Form.js'
import data from './data.json'


const dataLen = data.length


//data
describe("Number test", () => {
  test("Number of starting tasks is 3", () => {
    expect(dataLen).toBe(3)
  })
  
  test("Eat is already completed", () => {
    expect(data[0].completed).toBe(true)
  })
  
  test("Sleep is a starting task", () => {
    expect(data).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          "name": "Sleep"
        })
      ])
    )
  })

})

//Form
describe("Form snapshot test", () => {
  test('Form snapshot test', () => {
    const component = renderer.create(<Form />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})







