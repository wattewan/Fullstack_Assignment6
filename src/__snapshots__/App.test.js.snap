// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot test Form snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <h2
    className="label-wrapper"
  >
    <label
      className="label__lg"
      htmlFor="new-todo-input"
    >
      What needs to be done?
    </label>
  </h2>
  <input
    autoComplete="off"
    className="input input__lg"
    id="new-todo-input"
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn__primary btn__lg"
    type="submit"
  >
    Add
  </button>
</form>
`;
